impl/src/main/java/com/okta/sdk/impl/ds/DefaultDataStore.java

```diff
@@ -278,55 +275,52 @@ private ResourceAction getPostAction(ResourceDataRequest request, Response respo
-                if (!create) {
-                    method = HttpMethod.PUT;
-                }
+            // if this is an Okta user, we must use a PUT and not a POST
+            HttpMethod method = HttpMethod.POST;
+            if (!create) {
+                method = HttpMethod.PUT;
+            }
 
-                InputStream body;
-                long length = 0;
-                if (resource instanceof VoidResource) {
-                    body = new ByteArrayInputStream(new byte[0]);
-                } else {
-                    ByteArrayOutputStream bodyOut = new ByteArrayOutputStream();
-                    mapMarshaller.marshal(bodyOut, req.getData());
-                    body = new ByteArrayInputStream(bodyOut.toByteArray());
-                    length = bodyOut.size();
-                }
+            InputStream body;
+            long length = 0;
+            if (resource instanceof VoidResource) {
+                body = new ByteArrayInputStream(new byte[0]);
+            } else {
+                ByteArrayOutputStream bodyOut = new ByteArrayOutputStream();
+                mapMarshaller.marshal(bodyOut, req.getData());
+                body = new ByteArrayInputStream(bodyOut.toByteArray());
+                length = bodyOut.size();
+            }
 
-                Request request = new DefaultRequest(method, href, qs, httpHeaders, body, length);
-
-                Response response = execute(request);
-                Map<String, Object> responseBody = getBody(response);
-
-                if (Collections.isEmpty(responseBody)) {
-                    // Fix for https://github.com/stormpath/stormpath-sdk-java/issues/218
-                    // Okta response with 200 for deactivate requests (i.e. /api/v1/apps/<id>/lifecycle/deactivate)
-                    if (response.getHttpStatus() == 202
-                            || response.getHttpStatus() == 200
-                            || response.getHttpStatus() == 204) {
-                        //202 means that the request has been accepted for processing, but the processing has not been completed. Therefore we do not have a response body.
-                        responseBody = java.util.Collections.emptyMap();
-                    } else {
-                        throw new IllegalStateException("Unable to obtain resource data from the API server.");
-                    }
-                }
+            Request request = new DefaultRequest(method, href1, qs1, httpHeaders, body, length);
 
-                ResourceAction responseAction = getPostAction(req, response);
+            Response response = execute(request);
+            Map<String, Object> responseBody = getBody(response);
 
-                return new DefaultResourceDataResult(responseAction, uri, returnType, responseBody);
+            if (Collections.isEmpty(responseBody)) {
+                // Fix for https://github.com/stormpath/stormpath-sdk-java/issues/218
```

Comment:
Since this is licensed to Stormpath, we need to update the header to denote that they were changed.

See the [LICENSE](https://github.com/okta/okta-sdk-java/blob/5505055ee0974adf6f85b0d0efbe3fd5ad308401/LICENSE#L91).

Reply:
see: https://github.com/okta/okta-sdk-java/blob/master/THIRD-PARTY-NOTICES#L116

Reply:
IIRC including a notice to the `THIRD-PARTY-NOTICES` file doesn't indicate that you've modified the contents of the file.

@robertjd Do you recall what Legal requires us to do?

Reply:
;) https://github.com/okta/okta-sdk-java/blob/master/THIRD-PARTY-NOTICES#L105

Reply:
As mentioned, we should include the following on every modified file:
> Modifications Copyright (C) 2018 Okta, Inc. and/or its affiliates.

Lets pick this up that change on another PR